Error compiling on MacOSX with latest XCode

```
[ 14%] Building CXX object tests/src/Exceptions/ForeignThread/CMakeFiles/ForeignThreadExceptionsNative.dir/ForeignThreadExceptionsNative.cpp.o
In file included from __ROOT_GIT_REPO__/coreclr/tests/src/Exceptions/ForeignThread/ForeignThreadExceptionsNative.cpp:9:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/thread:97:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__mutex_base:17:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__threading_support:156:1: error: unknown type name
      'mach_port_t'
mach_port_t __libcpp_thread_get_port();
^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__threading_support:300:1: error: unknown type name
      'mach_port_t'
mach_port_t __libcpp_thread_get_port() {
^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__threading_support:301:12: error: use of undeclared
      identifier 'pthread_mach_thread_np'
    return pthread_mach_thread_np(pthread_self());
           ^
3 errors generated.
make[2]: *** [tests/src/Exceptions/ForeignThread/CMakeFiles/ForeignThreadExceptionsNative.dir/ForeignThreadExceptionsNative.cpp.o] Error 1
make[1]: *** [tests/src/Exceptions/ForeignThread/CMakeFiles/ForeignThreadExceptionsNative.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

I've substituted __ROOT_GIT_REPO__ for the root path to my git clone of this repo. Building from current master (which at time of submission of this issue was commit cbcdbd25e74ff9d963eafa202dd63504ca537f7e )

--- a/tests/src/Exceptions/ForeignThread/CMakeLists.txt
+++ b/tests/src/Exceptions/ForeignThread/CMakeLists.txt
@@ -4,6 +4,11 @@ project (ForeignThreadExceptionsNative)
 
 include_directories(${INC_PLATFORM_DIR})
 
+if(CLR_CMAKE_PLATFORM_DARWIN)
+  # Enable non-POSIX pthreads APIs, which by default are not included in the pthreads header
+  add_definitions(-D_DARWIN_C_SOURCE)
+endif(CLR_CMAKE_PLATFORM_DARWIN)
+
 set(SOURCES ForeignThreadExceptionsNative.cpp)
 
 # add the executable

